### PR TITLE
implemented purchase model, serializer, view

### PR DESCRIPTION
**Pull request by sanichdaniel**

### CODE REVIEW

File: transaction/models.py [hunk 1]
```diff
@@ -13,9 +13,27 @@ class Sale(models.Model):
...
+    title = models.CharField(max_length=100, default='')
+    content = models.CharField(max_length=2000, default='')
+    department = models.CharField(max_length=20, default='')
+    bookTitle = models.CharField(max_length=50, default='')
+    author = models.CharField(max_length=50, default='')
+    publisher = models.CharField(max_length=50, default='')
+    price = models.IntegerField()
+    isComplete = models.BooleanField(default=False)
 
-  class Meta:
```
Comment (yks1502) [inline transaction/models.py]: indent error

File: transaction/models.py [hunk 2]
```diff
@@ -13,9 +13,27 @@ class Sale(models.Model):
   publisher = models.CharField(max_length=50, default='')
   price = models.IntegerField()
   isComplete = models.BooleanField(default=False)
+class Meta:
```
Comment (yks1502) [inline transaction/models.py]: indent error

File: transaction/models.py [hunk 3]
```diff
@@ -13,9 +13,27 @@ class Sale(models.Model):
...
+    ordering = ('id',)
+
+def save(self, *args, **kwargs):
```
Comment (yks1502) [inline transaction/models.py]: indent error

File: transaction/models.py [hunk 4]
```diff
@@ -13,9 +13,27 @@ class Sale(models.Model):
...
+    isComplete = models.BooleanField(default=False)
 
-  class Meta:
+class Meta:
     ordering = ('id',)
 
-  def save(self, *args, **kwargs):
-    super(Sale, self).save(*args, **kwargs)
\ No newline at end of file
+def save(self, *args, **kwargs):
```
Comment (yks1502) [inline transaction/models.py]: indent error

File: transaction/serializers.py [hunk 5]
```diff
@@ -1,10 +1,16 @@
 from rest_framework import serializers
 from user.models import User
 from transaction.models import Sale
```
Comment (yks1502) [inline transaction/serializers.py]: merge two import lines

File: transaction/serializers.py [hunk 6]
```diff
@@ -1,10 +1,16 @@
...
+from transaction.models import Purchase
 
 class SaleSerializer(serializers.ModelSerializer):
-  
-  class Meta:
```
Comment (yks1502) [inline transaction/serializers.py]: indent error

File: transaction/urls.py [hunk 7]
```diff
@@ -4,5 +4,8 @@
 urlpatterns = [
   url(r'^sales$', views.SaleList.as_view()),
   url(r'^sale/(?P<pk>[0-9]+)$', views.SaleDetail.as_view()),
-  url(r'^complete/(?P<pk>[0-9]+)$', views.complete_sale),
-]
\ No newline at end of file
+  url(r'^complete_sale/(?P<pk>[0-9]+)$', views.complete_sale),
+  url(r'^purchases$', views.PurchaseList.as_view()),
+  url(r'^purchase/(?P<pk>[0-9]+)$', views.PurchaseDetail.as_view()),
+  url(r'^complete_purchase/(?P<pk>[0-9]+)$', views.complete_sale),
```
Comment (yks1502) [inline transaction/urls.py]: complete_sale -> complete_purchase

File: transaction/views.py [hunk 8]
```diff
@@ -1,6 +1,8 @@
 from django.shortcuts import render
 from transaction.models import Sale
```
Comment (yks1502) [inline transaction/views.py]: merge two import lines

File: transaction/views.py [hunk 9]
```diff
@@ -1,6 +1,8 @@
...
+from transaction.models import Purchase
 from transaction.serializers import SaleSerializer
```
Comment (yks1502) [inline transaction/views.py]: merge two import lines

File: transaction/views.py [hunk 10]
```diff
@@ -37,4 +39,38 @@ def complete_sale(request, pk):
...
+  serializer_class = PurchaseSerializer
+  permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+
+  def perform_create(self, serializer):
+    serializer.save(userId=self.request.user)
+
+class PurchaseDetail(generics.RetrieveUpdateDestroyAPIView):
+  queryset = Purchase.objects.all()
+  serializer_class = PurchaseSerializer
+  permissions_classes = (permissions.IsAuthenticatedOrReadOnly,)
```
Comment (yks1502) [inline transaction/views.py]: should fix permissions

Comment (sanichdaniel) [reply transaction/views.py]: fixed

File: transaction/views.py (hunk 9 above)
Comment (sanichdaniel) [reply transaction/views.py]: fixed

File: transaction/views.py (hunk 8 above)
Comment (sanichdaniel) [reply transaction/views.py]: fixed

File: transaction/urls.py (hunk 7 above)
Comment (sanichdaniel) [reply transaction/urls.py]: fixed

File: transaction/serializers.py (hunk 6 above)
Comment (sanichdaniel) [reply transaction/serializers.py]: fixed

File: transaction/serializers.py (hunk 5 above)
Comment (sanichdaniel) [reply transaction/serializers.py]: fixed

**Pull request merged**